src/sentry/static/sentry/app/views/organizationIntegrations/IntegrationDirectorySentryAppRow.tsx

```diff
@@ -22,42 +22,36 @@ type Props = {
   app: SentryApp;
   organization: Organization;
   install?: SentryAppInstallation;
-  isOnIntegrationPage: boolean;
   ['data-test-id']?: string;
 };
 
-export default class SentryApplicationRow extends React.PureComponent<Props> {
-  static propTypes = {
-    app: SentryTypes.SentryApplication,
-    organization: SentryTypes.Organization.isRequired,
-    install: PropTypes.object,
-    isOnIntegrationPage: PropTypes.bool,
-  };
-
+export default class IntegrationDirectorySentryAppRow extends React.PureComponent<Props> {
   get isInternal() {
     return this.props.app.status === 'internal';
   }
 
-  hideStatus() {
-    //no publishing for internal apps so hide the status on the developer settings page
-    return this.isInternal && !this.props.isOnIntegrationPage;
+  get isUnpublished() {
+    return this.props.app.status === 'unpublished';
   }
 
   renderStatus() {
-    const {app, isOnIntegrationPage} = this.props;
+    const {app} = this.props;
     const isInternal = this.isInternal;
     const status = this.installationStatus;
-    if (this.hideStatus()) {
-      return null;
-    }
-    if (isOnIntegrationPage) {
+    if (this.isInternal || this.isUnpublished) {
```

Comment:
@NisanthanNanthakumar I think having two different return statements is unnecessary. I think you could just have:
```
{!this.isPublished && <PublishStatus status={app.status} />}
```
In the other return statement (also need to write `isPublished`).

Reply:
@scefali not sure what you mean? Why is it unnecessary?

Reply:
@NisanthanNanthakumar Just mean you could write this more concisely